resources/assets/js/features/trip/components/Create/CreateTripContainer.js

```diff
@@ -0,0 +1,181 @@
+    }
+
+    getTime(start_at) {
+        if (!start_at) {
+            return {
+                start_at: null,
+                end_at: null
+            }
+        }
+
+        start_at = moment(start_at).unix();
+        let end_at = start_at + this.endTime;
+
+        return {
+            start_at: start_at,
+            end_at: end_at
+        }
+    }
+
+    onSubmit(e) {
+        e.preventDefault();
+
+        let time = this.getTime(e.target['start_at'].value);
+        let data = {
+            vehicle_id: e.target['vehicle_id'].value,
+            start_at: time.start_at,
+            end_at: time.end_at,
+            price: e.target['price'].value,
+            seats: e.target['seats'].value,
+            from: this.state.startPoint.place,
+            to: this.state.endPoint.place,
+        };
+
+        const validated = Validator.validate(createTripRules, data);
+
+        if (!validated.valid) {
+            this.setState({errors: validated.errors});
+            return;
+        }
+
+        this.setState({errors: {}});
+
+        securedRequest('post', '/api/trips/create', data).then((response) => {
+            this.setState({errors: {}});
+        }).catch((error) => {
+            this.setState({
+                errors: error.response.data
+            })
+        });
+    }
+
+    render() {
+        const placesCssClasses = {
+            root: 'form-group',
+            input: 'form-control',
+            autocompleteContainer: 'autocomplete-container'
+        };
+
+        const startPointProps = {
+            value: this.state.startPoint.address,
+            onChange: this.onChangeStartPoint.bind(this),
+        };
+
+        const endPointProps = {
+            value: this.state.endPoint.address,
+            onChange: this.onChangeEndPoint.bind(this),
+        };
+
+        return (
+            <div className="row">
+                <div className="col-sm-6">
+                    <CreateTripForm
+                        errors={this.state.errors}
+                        startPoint={startPointProps}
+                        endPoint={endPointProps}
+                        onSelectEndPoint={this.onSelectEndPoint.bind(this)}
+                        onSelectStartPoint={this.onSelectStartPoint.bind(this)}
+                        placesCssClasses={placesCssClasses}
+                        onSubmit={this.onSubmit.bind(this)}
+                    />
+                </div>
+                <div className="col-sm-6">
+                    <DirectionsMap title="Preview Trip"
+                                   from={CreateTripContainer.getCoordinatesFromPlace(this.state.startPoint.place)}
+                                   to={CreateTripContainer.getCoordinatesFromPlace(this.state.endPoint.place)}
+                                   endTime={this.setEndTime.bind(this)}
+                    />
+                </div>
+            </div>
+        );
+    }
+
+    static getCoordinatesFromPlace(place) {
```

Comment:
move this to GoogleMapService/GoogleMapManager or something like this^ that will know how to work with Google data